lumen/ai/views.py

```diff
@@ -243,6 +278,10 @@ async def render(self):
             return
 
         pipeline = self.component
+
+        # questionable if this is best way to handle memory
+        # does NOT work if user updates SQL from earlier in the history
+        self._memory["sql"] = self.spec
```

Comment:
I realized, if you edit the spec by hand/retry, the latter messages do not reference it because `memory` isn't updated, so this is my work around, but I don't think it's ideal

Reply:
Yeah, we need to handle replay. Technically editing for earlier outputs should either be locked or editing should replay all the subsequent steps.